src/python/pants/backend/codegen/protobuf/java/protobuf_gen.py

```diff
@@ -12,25 +12,24 @@
 from twitter.common.collections import OrderedSet
 
 from pants.backend.codegen.protobuf.java.java_protobuf_library import JavaProtobufLibrary
+from pants.backend.codegen.protobuf.subsystem.protoc import Protoc
 from pants.backend.jvm.targets.jar_library import JarLibrary
 from pants.backend.jvm.targets.java_library import JavaLibrary
 from pants.backend.jvm.tasks.jar_import_products import JarImportProducts
 from pants.base.build_environment import get_buildroot
 from pants.base.exceptions import TaskError
 from pants.base.workunit import WorkUnitLabel
-from pants.binaries.binary_util import BinaryUtil
 from pants.build_graph.address import Address
 from pants.fs.archive import ZIP
 from pants.task.simple_codegen_task import SimpleCodegenTask
-from pants.util.memo import memoized_property
 from pants.util.process_handler import subprocess
 
 
 class ProtobufGen(SimpleCodegenTask):
 
   @classmethod
   def subsystem_dependencies(cls):
-    return super(ProtobufGen, cls).subsystem_dependencies() + (BinaryUtil.Factory,)
+    return super(ProtobufGen, cls).subsystem_dependencies() + (Protoc,)
```

Comment:
Can we think of a reason not to scope this under the current task? Might someone want a different protoc here than in another task?

The "effect all consumers" option would end up looking the same, afaik. But you'd also be able to affect one consumer:
```
# affect all consumers
--protoc-version=
# affect one consumer
--protoc-gen-protoc-version=
```

Reply:
No good reason not to, except that we tend to use global instances everywhere unless we know we shouldn't. But we can start breaking that habit here...